lib/internal/util/comparisons.js

```diff
@@ -176,25 +207,28 @@ function innerDeepEqual(val1, val2, strict, memos) {
       return false;
     }
     return keyCheck(val1, val2, strict, memos, kIsArray, keys1);
-  }
-  if (val1Tag === '[object Object]') {
+  } else if (val1Tag === '[object Object]') {
     return keyCheck(val1, val2, strict, memos, kNoIterator);
-  }
-  if (isDate(val1)) {
-    if (DatePrototypeGetTime(val1) !== DatePrototypeGetTime(val2)) {
+  } else if (isDate(val1)) {
+    if (!isDate(val2) ||
```

Comment:
there's no failing tests that necessitate any change to the Date or RegExp or Error or boxed primitive logic - can you add those tests?

Reply:
Boxed primitives already have such test case.

I could add more tests for the other types but I actually like things like these to keep them around for coverage tasks at Code and Learn events. I am also fine to add the tests though, if you think that should be done in this PR.

Reply:
If there's that much of a shortage of tasks for those events, then so be it, but it seems unwise to make untested changes.

Reply:
I just added multiple tests and I indeed forgot to add one check :laughing: 
There are a couple more cases that are not yet completely covered: `Set`, `Map`, `ArrayBuffer` and non-native errors.